consul/acl.go

```diff
@@ -402,9 +442,15 @@ func (s *Server) filterACL(token string, subj interface{}) error {
 	case *structs.IndexedCheckServiceNodes:
 		filt.filterCheckServiceNodes(&v.Nodes)
 
+	case *structs.CheckServiceNodes:
+		filt.filterCheckServiceNodes(v)
+
```

Comment:
Nope - this one was new. I reused the CheckServiceNodes part for the query execution result.